xrpl/models/transactions/deposit_preauth.py

```diff
@@ -39,14 +45,86 @@ class DepositPreauth(Transaction):
                 "One of authorize and unauthorize must be set, not both."
             )
 
-        if not self.authorize and not self.unauthorize:
-            errors["DepositPreauth"] = "One of authorize and unauthorize must be set."
+        if (
+            self.authorize is None
+            and self.unauthorize is None
+            and self.authorize_credentials is None
+            and self.unauthorize_credentials is None
+        ):
+            errors["DepositPreauth"] = (
+                "Exactly one input parameter amongst authorize, unauthorize, "
+                + "authorize_credentials or unauthorize_credentials must be set. It is "
+                + "invalid if none of the params are specified."
+            )
+
+        # Filter for input parameters which are provided in the transaction
+        # Optional fields in the transaction model are set to None
+        params = [
+            self.authorize,
+            self.unauthorize,
+            self.authorize_credentials,
+            self.unauthorize_credentials,
+        ]
+        if sum(param is not None for param in params) > 1:
+            errors["DepositPreauth"] = (
+                "DepositPreauth txn accepts exactly one input amongst `authorize`, "
+                + "`unauthorize`, `authorize_credentials` and `unauthorize_credentials`."
+            )
+
+        def _validate_credentials_length(
+            credentials: List[Credential], field_name: str
+        ) -> None:
+            if len(credentials) == 0:
+                errors["DepositPreauth"] = f"{field_name} list cannot be empty. "
+            elif len(credentials) > MAX_CREDENTIAL_ARRAY_LENGTH:
+                errors["DepositPreauth"] = (
+                    f"{field_name} list cannot have more than "
+                    + str(MAX_CREDENTIAL_ARRAY_LENGTH)
+                    + " elements. "
+                )
+
+            if len(credentials) != len(set(credentials)):
+                errors["DepositPreauth"] = (
+                    f"{field_name} list cannot contain duplicate credentials."
+                )
+
+        if self.authorize_credentials is not None:
+            _validate_credentials_length(
+                self.authorize_credentials, "AuthorizeCredentials"
+            )
+
+        if self.unauthorize_credentials is not None:
+            _validate_credentials_length(
+                self.unauthorize_credentials, "UnauthorizeCredentials"
+            )
 
         return errors
+
+
+@require_kwargs_on_init
+@dataclass(frozen=True, **KW_ONLY_DATACLASS)
+class Credential(NestedModel):
+    """
+    An inner object representing individual element inside AuthorizeCredentials and
+    UnauthorizeCredentials array.
+    """
+
+    issuer: str = REQUIRED  # type: ignore
+    """The issuer of the credential."""
+
+    credential_type: str = REQUIRED  # type: ignore
+    """A (hex-encoded) value to identify the type of credential from the issuer."""
```

Comment:
```suggestion
    """A hex-encoded value to identify the type of credential from the issuer."""
```

Reply:
I copied this text verbatim from the docs website